From 8a521cb9e0a5f04e8905db45d8fbc629c4bf0428 Mon Sep 17 00:00:00 2001
From: nikolay serdjuk <nikolay.y.serdjuk@intel.com>
Date: Tue, 3 Dec 2013 11:31:25 +0700
Subject: Dalvik: Fix for circular dependency warning related to libdvm

BZ: 160629

In order to suppress the warning we need to exclude libpcgdvmjit.so from
dependencies for libdvm (the same goes for libcrash).
On the other hand we should leave libpcgdvmjit in the build process.
Therefore libpcgdvmjit is added in PRODUCT_PACKAGES variable
(see file device/intel/common/Android.mk).
But it doesnt mean that libpcgdvmjit will be always built because compilation
is controlled by WITH_PCG and TARGET_ARCH variables
(see file dalvik/Android.mk).
Depending on these variables the makefiles
vm/compiler/codegen/x86/pcg/Android.mk and libcrash/Android.mk are
included or excluded from the build process.
The following files were modified: dalvik/Android.mk, dalvik/vm/Android.mk and
device/intel/common/Android.mk.

A related patch can be found here http://android.intel.com:8080/#/c/148591/

Change-Id: I91eefd30b0c18eeb4ea4083906e62f7c30344339
Category: device enablement
Domain: AOSP.Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, need rework
Signed-off-by: nikolay serdjuk <nikolay.y.serdjuk@intel.com>
---
 Android.mk    |   14 +++++++-------
 vm/Android.mk |   11 -----------
 2 files changed, 7 insertions(+), 18 deletions(-)

diff --git a/Android.mk b/Android.mk
index dbb8524..2eb795f 100644
--- a/Android.mk
+++ b/Android.mk
@@ -94,15 +94,15 @@ subdirs := $(addprefix $(LOCAL_PATH)/,$(addsuffix /Android.mk, \
 		dx \
 		tools \
 		unit-tests \
-		libcrash \
 	))
 
-ifeq ($(WITH_JIT),true)
-  ifeq ($(WITH_PCG),true)
-    ifeq ($(TARGET_ARCH),x86)
-          subdirs += $(addprefix $(LOCAL_PATH)/,$(addsuffix /Android.mk, \
-                  vm/compiler/codegen/x86/pcg \
-              ))
+ifeq ($(TARGET_ARCH),x86)
+  subdirs += $(addprefix $(LOCAL_PATH)/,$(addsuffix /Android.mk, libcrash))
+  ifeq ($(WITH_JIT),true)
+    ifeq ($(WITH_PCG),true)
+      subdirs += $(addprefix $(LOCAL_PATH)/,$(addsuffix /Android.mk, \
+                 vm/compiler/codegen/x86/pcg \
+                 ))
     endif
   endif
 endif
diff --git a/vm/Android.mk b/vm/Android.mk
index df0800e..d2e29c7 100644
--- a/vm/Android.mk
+++ b/vm/Android.mk
@@ -55,19 +55,8 @@ endif
 
 # Overwrite default settings
 LOCAL_MODULE := libdvm
-ifeq ($(WITH_JIT),true)
-  ifeq ($(TARGET_ARCH),x86)
-    ifeq ($(WITH_PCG),true)
-      LOCAL_REQUIRED_MODULES += libpcgdvmjit
-    endif
-  endif
-endif
 LOCAL_CFLAGS += $(target_smp_flag)
 
-ifeq ($(TARGET_ARCH),x86)
-  LOCAL_REQUIRED_MODULES += libcrash
-endif
-
 # Define WITH_ADDRESS_SANITIZER to build an ASan-instrumented version of the
 # library in /system/lib/asan/libdvm.so.
 ifneq ($(strip $(WITH_ADDRESS_SANITIZER)),)
-- 
1.7.4.1

